### PR TITLE
[AIRFLOW-XXXX] Update LICENSE versions and remove old licenses

### PR DESCRIPTION
**Pull request by ryw**
We got what looks like a false-positive CVE alert from a customer using [Twistlock](https://www.twistlock.com/get-twistlock/) around jquery 2.1.4, which we're not using anymore, but was mentioned in LICENSE file:

image

This PR updates the versions in LICENSE, and in the process I discovered a number of referenced licenses for projects no longer in Airflow's dependency tree (from what I can tell - would love someone to double check my work).

---
Issue link: `Document only change, no JIRA issue`

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by ryw** [comment]
@ashb @potiuk would appreciate your review, make sure i didn't miss anything.

**Comment by ashb** [comment]
Twistlock really "detects" versions of code in use from what is mentioned in a license file? o_O

File: LICENSE [hunk 1]
```diff
@@ -229,36 +229,22 @@ MIT licenses
 The following components are provided under the MIT License. See project link for details.
 The text of each license is also included at licenses/LICENSE-[project].txt.
 
-    (MIT License) jquery v2.1.4 (https://jquery.org/license/)
-    (MIT License) dagre-d3 v0.6.1 (https://github.com/cpettitt/dagre-d3)
+    (MIT License) jquery v3.4.1 (https://jquery.org/license/)
+    (MIT License) dagre-d3 v0.8.5 (https://github.com/cpettitt/dagre-d3)
```
Comment (ashb) [inline LICENSE]: Yarn.lock has this as 0.6.4

Comment (ryw) [reply LICENSE]: good catch, i was looking at current version of dagre rather than dagre-d3

**Review by ryw — COMMENTED** [review]
> Twistlock really "detects" versions of code in use from what is mentioned in a license file? o_O

Unless we have jquery 2.1.4 actually tucked into the code somewhere — i can't find it

**Review by ashb — APPROVED** [review]

**Pull request merged**